src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/ProcessTests.cs

```diff
@@ -0,0 +1,244 @@
+using Microsoft.Diagnostics.Monitoring.UnitTests.Runners;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTests
+{
+    public class ProcessTests
+    {
+        private static readonly TimeSpan DefaultTimeout = TimeSpan.FromMinutes(1);
+
+        private readonly ITestOutputHelper _outputHelper;
+
+        public ProcessTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+        }
+
+        /// <summary>
+        /// Tests that a single process is discoverable by dotnet-monitor in connect mode.
+        /// </summary>
+        [Fact]
+        public Task SingleProcessConnectModeTest()
+        {
+            return SingleProcessCore(DiagnosticPortConnectionMode.Connect);
+        }
+
+#if NET5_0_OR_GREATER
+        /// <summary>
+        /// Tests that a single process is discoverable by dotnet-monitor in listen mode.
+        /// </summary>
+        [Fact]
+        public Task SingleProcessListenModeTest()
+        {
+            return SingleProcessCore(DiagnosticPortConnectionMode.Listen);
+        }
+#endif
+
+        /// <summary>
+        /// Tests that a single process is discoverable by dotnet-monitor.
+        /// </summary>
+        private async Task SingleProcessCore(DiagnosticPortConnectionMode monitorConnectionMode)
+        {
+            GenerateDiagnosticPortInfo(
+                monitorConnectionMode,
+                out DiagnosticPortConnectionMode appConnectionMode,
+                out string diagnosticPortPath);
+
+            await using MonitorRunner toolRunner = new(_outputHelper);
+            toolRunner.ConnectionMode = monitorConnectionMode;
+            toolRunner.DiagnosticPortPath = diagnosticPortPath;
+            toolRunner.DisableAuthentication = true;
+            await toolRunner.StartAsync(DefaultTimeout);
+
+            await Task.Delay(TimeSpan.FromSeconds(1));
+
+            using ApiClient client = new(_outputHelper, await toolRunner.GetDefaultAddressAsync(DefaultTimeout));
+
+            int appProcessId;
+            await using (AppRunner appRunner = new(_outputHelper))
+            {
+                appRunner.ConnectionMode = appConnectionMode;
+                appRunner.DiagnosticPortPath = diagnosticPortPath;
+                appRunner.ScenarioName = TestAppScenarios.SpinWait.Name;
+                await appRunner.StartAsync(DefaultTimeout);
+
+                appProcessId = appRunner.ProcessId;
+
+                await appRunner.SendStartScenarioAsync(DefaultTimeout);
+
+                await VerifyProcessAsync(client, await client.GetProcessesAsync(DefaultTimeout), appProcessId);
+
+                await EndSpinWaitScenarioAsync(appRunner);
+            }
+
+            await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
Maybe guard this around a method that we can change later + a TODO

Reply:
I fixed it to wait for the diagnostic pipes to no longer be reported.